### PR TITLE
get/import: more meaningful message on NoRemoteError

### LINKED ISSUE #2711: get/import: improve error message on no default remote
Since we allow importing/getting files that are not cached this error does not make much sense:

```
failed to import 'model.pkl' from '../test-import/'. - config file error: no remote specified. Setup default remote with
```

in a lot of cases.

### PR DESCRIPTION
**Pull request by pared**
* [x] ❗ Have you followed the guidelines in the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) list?

* [x] 📖 Check this box if this PR **does not** require [documentation](https://dvc.org/doc) updates, or if it does **and** you have created a separate PR in [dvc.org](https://github.com/iterative/dvc.org) with such updates (or at least opened an issue about it in that repo). Please link below to your PR (or issue) in the [dvc.org](https://github.com/iterative/dvc.org) repo.

* [x] ❌ Have you checked DeepSource, CodeClimate, and other sanity checks below? We consider their findings recommendatory and don't expect everything to be addresses. Please review them carefully and fix those that actually improve code or fix bugs.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Fixes to #2711

### CODE REVIEW

**Comment by shcheklein** [comment]
We have already a message that starts with `Setup default remote with ...` - do we keep it? @pared could you please show screens of how would it look like so that we can review?

File: dvc/exceptions.py [hunk 1]
```diff
@@ -328,3 +328,10 @@ def __init__(self, target_infos):
 
 class CollectCacheError(DvcException):
     pass
+
+
+class RemoteNotSpecifiedInTargetRepoError(DvcException):
+    def __init__(self, url):
+        super(RemoteNotSpecifiedInTargetRepoError, self).__init__(
+            "No remote is specified in target repository '{}'".format(url)
```
Comment (jorgeorpinel) [inline dvc/exceptions.py]: Specify "DVC remote"? Or "remote storage"? Git also has remotes.

**Comment by pared** [comment]
@shcheklein 
> We have already a message that starts with `Setup default remote with ...` - do we keep it? @pared could you please show screens of how would it look like so that we can review?

Reproduction script:
```
#!/bin/bash

rm -rf repo remote_repo storage git_repo
mkdir repo remote_repo storage git_repo

pushd remote_repo
git init >> /dev/null && dvc init -q

set -x
set -e
echo data >> data
dvc add data
git add .dvc/config data.dvc .gitignore
git commit -m "add data"

popd
pushd repo

git init >> /dev/null && dvc init -q
dvc import ../remote_repo data
```

Result:
https://asciinema.org/a/280490

File: dvc/exceptions.py (hunk 1 above)
Comment (pared) [reply dvc/exceptions.py]: Good point, Ill include that.

File: dvc/exceptions.py [hunk 2]
```diff
@@ -328,3 +328,10 @@ def __init__(self, target_infos):
 
 class CollectCacheError(DvcException):
     pass
+
+
+class RemoteNotSpecifiedInExternalRepoError(DvcException):
+    def __init__(self, url):
+        super(RemoteNotSpecifiedInExternalRepoError, self).__init__(
+            "No DVC remote is specified in target repository '{}'".format(url)
```
Comment (efiop) [inline dvc/exceptions.py]:
```suggestion
            "No DVC remote is specified in the target repository '{}'".format(url)
```

File: dvc/exceptions.py (hunk 1 above)
Comment (jorgeorpinel) [reply dvc/exceptions.py]: Thanks

File: dvc/external_repo.py [hunk 3]
```diff
@@ -19,7 +21,10 @@ def external_repo(url=None, rev=None, rev_lock=None, cache_dir=None):
 
     path = _external_repo(url=url, rev=rev_lock or rev, cache_dir=cache_dir)
     repo = Repo(path)
-    yield repo
+    try:
+        yield repo
+    except NoRemoteError:
+        raise RemoteNotSpecifiedInExternalRepoError(url)
```
Comment (efiop) [inline dvc/external_repo.py]: Just so we can get the traceback if we need it
```suggestion
        raise RemoteNotSpecifiedInExternalRepoError(url, cause=exc)
```

**Pull request merged**